src/utils/prelude.rs

```diff
@@ -25,12 +25,18 @@ pub(in crate) use ::core::{
         Not as _,
     },
 };
-#[cfg(not(target_arch = "wasm32"))]
-pub(in crate) use ::libc::size_t;
 
-#[cfg(target_arch = "wasm32")]
-#[allow(bad_style)]
-pub(in crate) type size_t = u32;
+cfg_if::cfg_if! {
+    if #[cfg(not(feature = "std"))] {
+        #[allow(bad_style)]
+        pub(in crate) type size_t = usize;
+    } else if #[cfg(target_arch = "wasm32")] {
+        #[allow(bad_style)]
+        pub(in crate) type size_t = u32;
```

Comment:
Let's keep the `no_std` & Wasm unified, since conceptually they're very much alike (Wasm not being `no-std` is a lie):

```suggestion
    if #[cfg(any(
        not(feature = "std"),
        target_arch = "wasm32",
    ))] {
        #[allow(bad_style)]
        pub(in crate) type size_t = usize;
```